File: Math/1492_the_kth_factor_of_n.cpp
class Solution {
public:
    int kthFactor(int n, int k) {
        int count = 0;
        for(int i = 1; i < n / 2 + 1; i += 1) {
            if(n % i == 0) {
                count += 1;
                if(count == k) return i;
            }
        }
        if(count + 1 == k) return n;
        return -1;
    }
};

File: Math/1492_the_kth_factor_of_n.py
class Solution:
    def kthFactor(self, n: int, k: int) -> int:
        if k == 1:
            return 1
        k -= 1

        for i in range(2, n // 2 + 1):
            if n % i == 0:
                k -= 1      
                if k == 0:
                    return i

        if k == 1:
            return n

        return -1
    
# revised - 02/19/2025